sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/FileSystemCertificates.java

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import java.io.InputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.FileInputStream;
+import java.io.File;
+import java.security.Key;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Arrays;
+
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.WARNING;
+
+/**
+ * Store certificates loaded from file system.
+ */
+final class FileSystemCertificates implements AzureCertificates {
+
+    /**
+     * Stores the logger.
+     */
+    private static final Logger LOGGER = Logger.getLogger(FileSystemCertificates.class.getName());
+
+    private final Map<String, List<String>> fileSystemAlias = new HashMap<>();
+
+    /**
+     * Stores the file system certificates by alias.
+     */
+    private final Map<String, Certificate> certificates = new HashMap<>();
+
+    /**
+     * Stores the file system certificate keys by alias.
+     */
+    private final Map<String, Key> certificateKeys = new HashMap<>();
+
+    private final List<String> certPaths;
+
+    private static volatile FileSystemCertificates fileSystemCertificatesInstance;
+
+    /**
+     * Get instance
+     * @return FileSystemCertificates instance
+     */
+    static FileSystemCertificates getInstance(List<String> certPaths) {
```

Comment:
If user use different `certPaths`, it returns same `FileSystemCertificates`, it's not what user want, I think.